aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/NativeClient.java

```diff
@@ -44,10 +46,25 @@
             initializeClient(userAgent.getBytes());
     }
 
+    /**
+     * @return a string describing the detected architecture the RIC is executing on
+     * @throws RuntimeException
+     */
+    static String getArchIdentifier() {
+        String arch = System.getProperty("os.arch");
+
+        if (arch.matches("^(x8664|amd64|ia32e|em64t|x64|x86_64|x8632|x86|i[3-6]86|ia32|x32)$")) {
+            return "x86";
+        } else if (arch.matches("^(armeabi.*|arm64.*|aarch64.*|arm)$")) {
+            return "arm";
+        }
+
+        throw new RuntimeException("architecture not supported: " + arch);
```

Comment:
let us [not throw](https://wiki.sei.cmu.edu/confluence/display/java/ERR07-J.+Do+not+throw+RuntimeException%2C+Exception%2C+or+Throwable) RuntimeException

Reply:
Suggest wrapping in new exception eg. `UnknownPlatformException`